cro-clib/src/basic_wallet.rs

```diff
@@ -0,0 +1,166 @@
+use crate::types::{CroAddress, CroAddressPtr, CroResult};
+use chain_core::init::address::RedeemAddress;
+
+use chain_core::state::account::StakedStateAddress;
+use chain_core::tx::data::address::ExtendedAddr;
+use client_common::MultiSigAddress;
+use client_common::{PrivateKey, PublicKey};
+use std::ptr;
+
+/// # Safety
+unsafe extern "C" fn do_cro_basic_create_staking_address(
```

Comment:
why `do` ? 

perhaps comment should mention that this creates a non-HD from the provided private key?


And does this need to be unsafe if it's no public and only internal function?

Reply:
it's for handling rust errors

Reply:
it's for handling rust errors , because interface function cannot use rust type error handling.

Reply:
but this function isn't public, it's not exposed -- it could just be a normal Rust function?

Reply:
yes, i agree, i'll change it